### ISSUE TITLE
Update middleware lib

### ISSUE THREAD

**Issue by samlaf** [Layr-Labs/incredible-squaring-avs#43]
We are currently on few months old version of middleware that doesn't have AvsDirectory as an example.
We need to update.

**Comment by wawrzek** [comment]
I tried to deploy the example to the current local deployment of the EigenLayer and it failed, so I hope that by middleware you also mean eignelayer-contrats.

The error I see is:

```
  ~/Etherum/iv1 : wn/add-test-avs ± > docker-compose logs avs-test
avs-test-1  | No files changed, compilation skipped
avs-test-1  | Traces:
avs-test-1  |   [21771599] → new IncredibleSquaringDeployer@0x5b73C5498c1E3b4dbA84de0F1833c4a029d90519
avs-test-1  |     └─ ← 108533 bytes of code
avs-test-1  |
avs-test-1  |   [985724] IncredibleSquaringDeployer::run()
avs-test-1  |     ├─ [0] VM::projectRoot() [staticcall]
avs-test-1  |     │   └─ ← "/eigenlayer/incredible-squaring-avs/contracts"
avs-test-1  |     ├─ [0] VM::toString(32382 [3.238e4]) [staticcall]
avs-test-1  |     │   └─ ← "32382"
avs-test-1  |     ├─ [0] VM::readFile("/eigenlayer/incredible-squaring-avs/contracts/script/output/32382/eigenlayer_deployment_output.json") [staticcall]
avs-test-1  |     │   └─ ← <file>
avs-test-1  |     ├─ [0] VM::parseJsonAddress("<stringified JSON>", ".addresses.strategyManager") [staticcall]
avs-test-1  |     │   └─ ← 0x75A5d120897A76224a3E8e62dC0F18cc808F739d
avs-test-1  |     ├─ [0] VM::parseJsonAddress("<stringified JSON>", ".addresses.delegation") [staticcall]
avs-test-1  |     │   └─ ← 0x30bdaE426d3CBD42e9d41D23958Fac6AD8310f81
avs-test-1  |     ├─ [0] VM::parseJsonAddress("<stringified JSON>", ".addresses.eigenLayerProxyAdmin") [staticcall]
avs-test-1  |     │   └─ ← 0x3e2AabB763F255CbB6a322DBe532192e120B5C6B
avs-test-1  |     ├─ [0] VM::parseJsonAddress("<stringified JSON>", ".addresses.eigenLayerPauserReg") [staticcall]
avs-test-1  |     │   └─ ← 0x3B92c21Ad01E7091513eAb0c26C28B8Ba1Ee21ab
avs-test-1  |     ├─ [0] VM::parseJsonAddress("<stringified JSON>", ".addresses.baseStrategyImplementation") [staticcall]
avs-test-1  |     │   └─ ← 0xF3073ab62A79c316D58556555771de5A906c7E54
avs-test-1  |     ├─ [0] VM::startBroadcast()
avs-test-1  |     │   └─ ← ()
avs-test-1  |     ├─ [264106] → new ERC20Mock@0x439a926961EFc90dFD19f952F79A2b07530059f9
avs-test-1  |     │   └─ ← 1319 bytes of code
avs-test-1  |     ├─ [582386] → new TransparentUpgradeableProxy@0x3a63ac6a25cb4ed2c85A8A986F4032003e5dC734
avs-test-1  |     │   ├─ emit Upgraded(implementation: 0xF3073ab62A79c316D58556555771de5A906c7E54)
avs-test-1  |     │   ├─ [99364] 0xF3073ab62A79c316D58556555771de5A906c7E54::initialize(1000000000000000000 [1e18], 100000000000000000000 [1e20], ERC20Mock: [0x439a926961EFc90dFD19f952F79A2b07530059f9], 0x3B92c21Ad01E7091513eAb0c26C28B8Ba1Ee21ab) [delegatecall]
avs-test-1  |     │   │   ├─ emit MaxPerDepositUpdated(previousValue: 0, newValue: 1000000000000000000 [1e18])
avs-test-1  |     │   │   ├─ emit MaxTotalDepositsUpdated(previousValue: 0, newValue: 100000000000000000000 [1e20])
avs-test-1  |     │   │   ├─ emit Paused(account: 0x123463a4B065722E99115D6c222f267d9cABb524, newPausedStatus: 0)
avs-test-1  |     │   │   ├─ emit PauserRegistrySet(pauserRegistry: 0x0000000000000000000000000000000000000000, newPauserRegistry: 0x3B92c21Ad01E7091513eAb0c26C28B8Ba1Ee21ab)
avs-test-1  |     │   │   ├─ emit Initialized(version: 1)
avs-test-1  |     │   │   └─ ← ()
avs-test-1  |     │   ├─ emit AdminChanged(previousAdmin: 0x0000000000000000000000000000000000000000, newAdmin: 0x3e2AabB763F255CbB6a322DBe532192e120B5C6B)
avs-test-1  |     │   └─ ← 2151 bytes of code
avs-test-1  |     ├─ [7563] 0x75A5d120897A76224a3E8e62dC0F18cc808F739d::addStrategiesToDepositWhitelist([0x3a63ac6a25cb4ed2c85A8A986F4032003e5dC734])
avs-test-1  |     │   ├─ [237] 0x61A84cd025ec5EEbEF2098C9d933A49332BE68B4::addStrategiesToDepositWhitelist([0x3a63ac6a25cb4ed2c85A8A986F4032003e5dC734]) [delegatecall]
avs-test-1  |     │   │   └─ ← EvmError: Revert
avs-test-1  |     │   └─ ← EvmError: Revert
avs-test-1  |     └─ ← EvmError: Revert
avs-test-1  |
avs-test-1  |
avs-test-1  | Error:
avs-test-1  | script failed: <empty revert data>
```

**Comment by samlaf** [comment]
@wawrzek what exactly is your setup? `make deploy-all-to-anvil-and-save-state` works right now afaik.
Also yes, middleware has eigenlayer-contracts as a dependency, so updating middleware will also update the eigenlayer-contracts version.

**Comment by wawrzek** [comment]
@samlaf - I started to build https://github.com/ivy-net/iv1 - which is a POS Ethereum DevNet in docker-composed with EigenLayer deployed into it. The problem is that I deploy the eignerlayer-contracts from dev, but incredible-squaring-avs is bind to the rather old commit:
```
commit aa8b38fce32534dfa9feb485d17b1f832de22bd3
Author: Samuel Laferriere <samlaf92@gmail.com>
Date:   Tue Dec 19 21:23:05 2023 -0500

    Fix m2 deploy from scratch script m2 mainnet (#379)
```


If you would like to try iv1, you can clone the repo; follow the quick start, and then, when it's deployed, run the following command from the incredible-squaring-avs repo.

```
 forge script script/IncredibleSquaringDeployer.s.sol --rpc-url http://localhost:8545 --broadcast --unlocked --sender 0x123463a4b065722e99115d6c222f267d9cabb524 -vvvv
```

An alternative is to look at the `wn/add-test-avs` branch, where I try to deploy the incredible-squaring-avs repo as an extra step.

**Comment by samlaf** [comment]
Fixed by #46 . Closing
